cartography/intel/aws/ec2/instances.py

```diff
@@ -65,85 +65,90 @@ def transform_ec2_instances(reservations: List[Dict[str, Any]], region: str, cur
+                    keypair_list.append({
+                        'KeyPairArn': key_pair_arn,
+                        'KeyName': key_name,
                         'InstanceId': instance_id,
-                        'SubnetId': subnet_id,
-                        'GroupId': security_group['GroupId'],
                     })
 
-            if 'BlockDeviceMappings' in instance and len(instance['BlockDeviceMappings']) > 0:
-                for mapping in instance['BlockDeviceMappings']:
-                    if 'VolumeId' in mapping['Ebs']:
-                        instance_ebs_volumes_list.append({
+                if instance.get("SecurityGroups"):
+                    for group in instance["SecurityGroups"]:
+                        sg_list.append(
+                            {
+                                'GroupId': group['GroupId'],
+                                'InstanceId': instance_id,
+                            },
+                        )
+
+                for network_interface in instance['NetworkInterfaces']:
+                    for security_group in network_interface.get('Groups', []):
+                        network_interface_list.append({
+                            'NetworkInterfaceId': network_interface['NetworkInterfaceId'],
+                            'Status': network_interface['Status'],
+                            'MacAddress': network_interface['MacAddress'],
+                            'Description': network_interface['Description'],
+                            'PrivateDnsName': network_interface.get('PrivateDnsName'),
+                            'PrivateIpAddress': network_interface['PrivateIpAddress'],
                             'InstanceId': instance_id,
-                            'VolumeId': mapping['Ebs']['VolumeId'],
-                            'DeleteOnTermination': mapping['Ebs']['DeleteOnTermination'],
-                            # 'SnapshotId': mapping['Ebs']['SnapshotId'],  # TODO check on this
+                            'SubnetId': subnet_id,
+                            'GroupId': security_group['GroupId'],
                         })
 
+                if 'BlockDeviceMappings' in instance and len(instance['BlockDeviceMappings']) > 0:
+                    for mapping in instance['BlockDeviceMappings']:
+                        if 'VolumeId' in mapping['Ebs']:
+                            instance_ebs_volumes_list.append({
+                                'InstanceId': instance_id,
+                                'VolumeId': mapping['Ebs']['VolumeId'],
+                                'DeleteOnTermination': mapping['Ebs']['DeleteOnTermination'],
+                                # 'SnapshotId': mapping['Ebs']['SnapshotId'],  # TODO check on this
+                            })
+            except Exception as e:
+                logger.error(f"Failed to process instance {instance_id}. Error: {e}")
+                raise e
```

Comment:
Added log to identify instance failing.

Reply:
Generally not the best practice to add an extra log. You can wrap the exception by another one to clarify the error.

```
except Exception as e:
  raise Exception(f"Failed to process instance {instance_id}", e)
```

Sometimes it's also preferred to have a named exception but i think it might not be necessary in this case.

Reply:
Yeah, I can wrap it to extend the data with the instance id.